mautrix_telegram/portal.py

```diff
@@ -2318,6 +2341,19 @@ async def _handle_telegram_document(
                 "image/": MessageType.IMAGE,
             }.get(info.mimetype[:6], MessageType.FILE),
         )
+        if attrs.is_audio:
+            content["org.matrix.msc1767.file"] = {
+                "url": content.url,
+                "name": content.body,
+                **(content.file.serialize() if content.file else {}),
+                **(content.info.serialize() if content.info else {}),
+            }
+            content["org.matrix.msc1767.audio"] = {
+                "duration": attrs.duration,
+                "waveform": [int(x) for x in attrs.waveform] if attrs.waveform else None,
```

Comment:
This should never be none? (if the goal is to not send empty arrays, then the field should probably be excluded completely when empty)

```suggestion
                "waveform": [int(x) for x in attrs.waveform],
```